api/app/controllers/spree/api/v2/base_controller.rb

```diff
@@ -41,6 +42,10 @@ def find_spree_current_order
         def current_currency
           spree_current_store.default_currency || Spree::Config[:currency]
         end
+
+        def rescue_record_not_found(exception)
+          render json: { error: exception.message }, status: :not_found
```

Comment:
`404`? 
Let's be consistent with statuses since we've agreed to use numeric values as opposed to symbols.